includes/gateway/class-omise-payment-installment.php

```diff
@@ -0,0 +1,262 @@
+<?php
+defined( 'ABSPATH' ) or die( 'No direct script access allowed.' );
+
+function register_omise_installment() {
+	require_once dirname( __FILE__ ) . '/class-omise-payment.php';
+
+	if ( ! class_exists( 'WC_Payment_Gateway' ) ) {
+		return;
+	}
+
+	if ( class_exists( 'Omise_Payment_Installment' ) ) {
+		return;
+	}
+
+	/**
+	 * @since 3.4
+	 */
+	class Omise_Payment_Installment extends Omise_Payment {
+		public function __construct() {
+			parent::__construct();
+
+			$this->id                 = 'omise_installment';
+			$this->has_fields         = true;
+			$this->method_title       = __( 'Omise Installment', 'omise' );
+			$this->method_description = wp_kses(
+				__( 'Accept payment through <strong>Installment</strong> via Omise payment gateway.', 'omise' ),
+				array( 'strong' => array() )
+			);
+
+			$this->init_form_fields();
+			$this->init_settings();
+
+			$this->title       = $this->get_option( 'title' );
+			$this->description = $this->get_option( 'description' );
+
+			add_action( 'wp_enqueue_scripts', array( $this, 'omise_assets' ) );
+			add_action( 'woocommerce_update_options_payment_gateways_' . $this->id, array( $this, 'process_admin_options' ) );
+			add_action( 'woocommerce_api_' . $this->id . '_callback', array( $this, 'callback' ) );
+		}
+
+		/**
+		 * @see WC_Settings_API::init_form_fields()
+		 * @see woocommerce/includes/abstracts/abstract-wc-settings-api.php
+		 */
+		public function init_form_fields() {
+			$this->form_fields = array(
+				'enabled' => array(
+					'title'   => __( 'Enable/Disable', 'omise' ),
+					'type'    => 'checkbox',
+					'label'   => __( 'Enable Omise Installment Payment', 'omise' ),
```

Comment:
'Enable Omise Installment Payments'

Reply:
@jonrandy ✋ wonder why we use `payments` instead of `payment`? Is it because there are multiple installment providers?

Reply:
because there are always multiple payments when you pay this way, never just one